packages/@aws-cdk/aws-elasticloadbalancingv2/lib/alb/application-listener-action.ts

```diff
@@ -0,0 +1,446 @@
+        targetGroupStickinessConfig: options.stickinessDuration ? {
+          durationSeconds: options.stickinessDuration.toSeconds(),
+          enabled: true,
+        } : undefined,
+      },
+    });
+  }
+
+  /**
+   * Return a fixed response
+   */
+  public static fixedResponse(statusCode: number, options: FixedResponseOptions = {}): ListenerAction {
+    return new ListenerAction({
+      type: 'fixed-response',
+      fixedResponseConfig: {
+        statusCode: Tokenization.stringifyNumber(statusCode),
+        contentType: options.contentType,
+        messageBody: options.messageBody,
+      },
+    });
+  }
+
+  /**
+   * Redirect to a different URI
+   *
+   * A URI consists of the following components:
+   * protocol://hostname:port/path?query. You must modify at least one of the
+   * following components to avoid a redirect loop: protocol, hostname, port, or
+   * path. Any components that you do not modify retain their original values.
+   *
+   * You can reuse URI components using the following reserved keywords:
+   *
+   * - `#{protocol}`
+   * - `#{host}`
+   * - `#{port}`
+   * - `#{path}` (the leading "/" is removed)
+   * - `#{query}`
+   *
+   * For example, you can change the path to "/new/#{path}", the hostname to
+   * "example.#{host}", or the query to "#{query}&value=xyz".
+   */
+  public static redirect(options: RedirectOptions): ListenerAction {
+    if ([options.host, options.path, options.port, options.protocol, options.query].findIndex(x => x !== undefined) === -1) {
+      throw new Error('To prevent redirect loops, set at least one of \'protocol\', \'host\', \'port\', \'path\', or \'query\'.');
+    }
+
+    return new ListenerAction({
+      type: 'redirect',
+      redirectConfig: {
+        statusCode: options.permanent ? 'HTTP_301' : 'HTTP_302',
+        host: options.host,
+        path: options.path,
+        port: options.port,
+        protocol: options.protocol,
+        query: options.query,
+      },
+    });
+  }
+
+  /**
+   * Create an instance of ListenerAction
+   *
+   * The default class should be good enough for most cases and
+   * should be created by using one of the static factory functions,
+   * but allow overriding to make sure we allow flexibility for the future.
+   */
+  protected constructor(private readonly actionJson: CfnListener.ActionProperty, protected readonly next?: ListenerAction) {
+  }
+
+  /**
+   * Render the actions in this chain
+   */
+  public renderActions(): CfnListener.ActionProperty[] {
+    return this.renumber([this.actionJson, ...this.next?.renderActions() ?? []]);
+  }
+
+  /**
+   * Called when the action is being used in a listener
+   */
+  public bindToListener(scope: Construct, listener: IApplicationListener, associatingConstruct?: IConstruct) {
+    // Empty on purpose
+    Array.isArray(scope);
+    Array.isArray(listener);
+    Array.isArray(associatingConstruct);
+  }
```

Comment:
Why not just call this `bind()` like we usually do in other places?

Also why not `abstract`?

https://github.com/aws/aws-cdk/search?q=%22bind%22&unscoped_q=%22bind%22

Reply:
Because it is empty on purpose, for most of the actions.